lasagne/regularization.py

```diff
@@ -1,15 +1,131 @@
 import theano.tensor as T
+from .layers import Layer, get_all_params
 
-from . import layers
 
+def l1(x):
+    """Computes the L1 norm of a tensor
 
-def l2(layer, include_biases=False):
-    if include_biases:
-        all_params = layers.get_all_params(layer)
-    else:
-        all_params = layers.get_all_non_bias_params(layer)
+    Parameters
+    ----------
+    x : Theano tensor
 
-    return sum(T.sum(p**2) for p in all_params)
+    Returns
+    -------
+    Theano scalar
+        l1 norm (sum of absolute values of elements)
+    """
+    return T.sum(abs(x))
 
 
-# TODO: sparsity regularization
+def l2(x):
+    """Computes the squared L2 norm of a tensor
+
+    Parameters
+    ----------
+    x : Theano tensor
+
+    Returns
+    -------
+    Theano scalar
+        squared l2 norm (sum of squared values of elements)
+    """
+    return T.sum(x**2)
+
+
+def apply_penalty(tensor_or_tensors, penalty, **kwargs):
+    """
+    Computes the total cost for applying a specified penalty
+    to a tensor or group of tensors.
+
+    Parameters
+    ----------
+    tensor_or_tensors : Theano tensor or list of tensors
+    penalty : callable
+    kwargs : keyword arguments passed to penalty
+
+    Returns
+    -------
+    Theano scalar
+        a scalar expression for the total penalty cost
+    """
+    try:
+        return sum(penalty(x, **kwargs) for x in tensor_or_tensors)
+    except (TypeError, ValueError):
+        return penalty(tensor_or_tensors, **kwargs)
```

Comment:
What case is the `ValueError` for? Isn't the `TypeError` enough to distinguish between iterable and non-iterable `tensor_or_tensors`?